### DISCUSSION TITLE
How to lock/prevent/disable `account.create()`?

### DISCUSSION THREAD

**Discussion by lawipac** [appwrite/appwrite#6049] [Q&A] [answered]
Hi Everyone

For a webAPP, API `account.create()`  can create as many accounts as it wants, how to prevent it from happening?  To my best understanding, it requires a webpage hosted in an authorized domain-name.  using CORS to prevent unauthorized js code. 

but my question is people can still copy the project ID and setup a local webserver, using /etc/hosts to trick the browser to think local server is the legit FQDN specified by CORS.

Then, a malicious can create many zombie accounts, or unverified accounts to pollute the user database and possibly block real user from registering by using their email address. 

Any way to restrict `account.create()` from being called just by knowing project ID?

**Comment by stnguyen90** [comment, marked as answer]
The best way to prevent account creations is to set the Users limit to a low number like 1. With this set to a low number, Appwrite will return an error when an attempt to create a new user is made, such as in the account create endpoint:

https://github.com/appwrite/appwrite/blob/fbd277f6aa0ff161838a60e79b950e5a3f58e3f6/app/controllers/api/account.php#L99-L103

You can find this setting in the Appwrite Counsole > Your Project > Auth > Security > Users limit:

<img width="1407" alt="image" src="https://github.com/appwrite/appwrite/assets/1477010/69d14b1d-ef2b-4983-9be1-2da2c3c6eecd">

Even though this is set, you can still create users via the Appwrite Console or a server SDK (and API key).
